tests/EnumTest.php

```diff
@@ -38,42 +39,44 @@ public function testGetKey()
     }
 
     /**
-     * @expectedException \UnexpectedValueException
+     * @dataProvider invalidValueProvider
      */
-    public function testInvalidValueString()
+    public function testCreatingEnumWithInvalidValue($value)
     {
-        new EnumFixture("test");
-    }
+        $this->setExpectedException(
+            '\UnexpectedValueException',
+            'Value \'' . $value . '\' is not part of the enum MyCLabs\Tests\Enum\EnumFixture'
+        );
 
-    /**
-     * @expectedException \UnexpectedValueException
-     */
-    public function testInvalidValueInt()
-    {
-        new EnumFixture(1234);
+        new EnumFixture($value);
     }
 
     /**
-     * @expectedException \UnexpectedValueException
+     * Contains values not existing in EnumFixture
+     * @return array
      */
-    public function testInvalidValueEmpty()
-    {
-        new EnumFixture(null);
+    public function invalidValueProvider() {
+        return [
+            "string" => ['test'],
+            "int" => [1234],
+        ];
     }
 
     /**
      * __toString()
+     * @dataProvider toStringProvider
      */
-    public function testToString()
+    public function testToString($expected, $enumObject)
     {
-        $value = new EnumFixture(EnumFixture::FOO);
-        $this->assertEquals(EnumFixture::FOO, (string) $value);
-
-        $value = new EnumFixture(EnumFixture::BAR);
-        $this->assertEquals(EnumFixture::BAR, (string) $value);
+        $this->assertSame($expected, (string) $enumObject);
+    }
 
-        $value = new EnumFixture(EnumFixture::NUMBER);
-        $this->assertEquals((string) EnumFixture::NUMBER, (string) $value);
+    public function toStringProvider() {
+        return [
+            [EnumFixture::FOO, new EnumFixture(EnumFixture::FOO)],
+            [EnumFixture::BAR, new EnumFixture(EnumFixture::BAR)],
+            [(string) EnumFixture::NUMBER, new EnumFixture(EnumFixture::NUMBER)],
+        ];
```

Comment:
We have to keep BC with PHP 5.3 (yeah I know -_-)

Tests are failing because of that.